### ISSUE TITLE
$gt selector doesn't work correctly with dates

### ISSUE THREAD

**Issue by wujitouch** [nolanlawson/pouchdb-find#38]
I'm not sure if I have it set up wrong or what but when the $gt selector doesn't seem to work with dates.

Data:
{
    docType: 'log',
    logDate: 1432823742926
}

Query:
var start = (new Date()).getTime();
db.find({
    selector: { docType: 'log', startTime: { $gt: start } }
});

The above query doesn't returns no results with correct values for start and logDate. I also tried this without using epoch time and letting the date get serialized to ISO format and this doesn't work either. For some reason, it works if I use $gte

Edit:
I've created a plunker showing the issue here:
http://plnkr.co/edit/FU3SFp3E30ZTOElTbNSU

**Comment by nolanlawson** [comment]
Can you provide a live test case? Just makes it easier to confirm what's going on. A JSBin or even just a Node script would be fine. :)

**Comment by wujitouch** [comment]
Hi Nolan, thanks for the quick response. I created a plunker here, just in case you didn't see the edit:
http://plnkr.co/edit/FU3SFp3E30ZTOElTbNSU

**Comment by wujitouch** [comment]
I just noticed that $gte doesn't work either when combined with $lt or $lte. It seems like it completely ignores $gt/$gte

**Comment by wujitouch** [comment]
Hi Nolan,

I'm sure you're very busy but have you had a chance to look at my plunk?

Thanks

**Comment by nolanlawson** [comment]
Sorry, I looked at it just now.

```js
db.post({ docType: 'log', logDate: new Date('2015-05-24')});
```

You cannot save Dates in PouchDB. IndexedDB will save the dates for you without emitting a warning, but it is a lie because the other adapters (and CouchDB) cannot, and unexpected things will happen when you do so.

You need to do:

```js
db.post({ docType: 'log', logDate: new Date('2015-05-24').toISOString()});
```

**Comment by wujitouch** [comment]
Thanks for taking a look. I changed everything to use only strings and $gt and $gte still don't work correctly. See the updated plunk please. You can see the output in the console. I'm using Firefox, if that matters.

In the console, these two queries show something's wrong:

"-- querying docs logDate > 2015-05-25T00:00:00.000Z"
""
"-- querying docs logDate >= 2015-05-25T00:00:00.000Z and <= 2015-05-26T00:00:00.000Z" 
"2015-05-24T00:00:00.000Z, 2015-05-25T00:00:00.000Z, 2015-05-26T00:00:00.000Z, "

The first query should return 2 docs (26th and 27th). The second should only return doc with logDate 26th

**Comment by nolanlawson** [comment]
Okay, it does indeed look like a bug. I am very busy right now and will probably not be able to get to it for several weeks at the maximum.

If you would like to contribute the fix, you just need to write a test and fix it. :) The tests are described in [CONTRIBUTING.md](https://github.com/nolanlawson/pouchdb-find/blob/master/CONTRIBUTING.md) and are pretty easy to set up. Basically just check out the code, connect it to your Cloudant, and then run `npm test`.

**Comment by nolanlawson** [comment]
fixed by https://github.com/nolanlawson/pouchdb-find/pull/44

**Comment by nolanlawson** [comment]
fixed in 0.3.5
